app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,87 @@
+class OrdersController < ApplicationController
+  before_action :require_login, only: [:show_seller_orders]
+
+  def show
+    @orders = Order.find(current_order.id).orderitems
+  end
+
+  def show_seller_orders
+  	@user = User.find(current_user.id)
+    @user_orders_hash = Orderitem.where(user: current_user).group_by(&:order_id)
+    @revenue = @user.revenue
+    @completed_revenue = @user.revenue_by_status("Completed")
+    @pending_revenue = @user.revenue_by_status("Pending")
+    @completed_count = @user.order_by_status("Completed")
+    @pending_count = @user.order_by_status("Pending")
+  end
+
+  def order_deets
+    @user_orders_hash = Orderitem.where(user: current_user).group_by(&:order_id)
+    @order = Order.find(params[:order_id])
+  end
+
+
+  def edit
+    @order  = Order.find(params[:id])
+    @orderitems = Order.find(current_order.id).orderitems
+  end
+
+  def update
+    @order = current_order
+    @orderitems = @order.orderitems
+    continue = remove_items_from_stock(@orderitems)
+    unless continue == false
+      @order.update(order_update_params[:order])
+    end
+    if @order.status == "Completed"
+      redirect_to order_confirmation_path(@order.id)
+    else
+      flash[:notice] = "Sorry! An item you wanted is out of stock. Check to see if you have duplicate items in your cart."
+      redirect_to edit_order_path(current_order.id)
+    end
+  end
+
+  def checkout
+    @order = current_order
+    if @order.orderitems.count == 0
+      redirect_to edit_order_path(current_order.id), alert: "Please add items to your cart!"
+    end
+  end
+
+  def confirmation
+    @order = current_order
+    @orderitems = @order.orderitems
+    session.delete :order_id
+    order = Order.create
+    order.update(status: "Pending")
+    session[:order_id] = order.id
+  end
+
+
+  private
+
+  def remove_items_from_stock(items)
+    items.each do |item|
+      product = item.product
+      quantity_being_bought = item.quantity
+      available_quantity = product.quantity
+      new_quantity = available_quantity - quantity_being_bought
+      if new_quantity >= 0
+        product.update(quantity: new_quantity)
+      else
+        return false
+      end
+    end
+  end
```

Comment:
We wondered about that and now that we know more about where things should go you're right - this would be great to move to the model!